doc/kazoo/cluster-guide.md

```diff
@@ -1,9 +1,277 @@
-## Key Dedicated
+## Cluster Guide
 
 Name: **Kazoo Dedicated Cluster Guide**
 
 Description: *A guide for installing, configuring and managing your own dedicated Kazoo cluster.*
 
+Let's assume the following 7 server clusters in 2 zones.
+### Abbreviation
+bc = Bigcouch  
+fs = Freeswitch    
+mq = RabbitMQ  
+kz = Kazoo  
+ka = Kamailio  
 
-*Created by
-dschreiber (Jan 04, 2014)
+### IP addressing scheme  
+10.100 = zone 100  
+10.200 = zone 200  
+10.x00.10.x = Bigcouch  
+10.x00.20.x = Freeswitch  
+10.x00.30.x = RabbitMQ  
+10.x00.40.x = Kazoo  
+10.x00.50.x = Kamailio  
+
+### Server hostnames and IP addresses
+Using the abbreviation and IP addressing scheme above.
+
+| ZONE 1 | ZONE 2 |
+| ---------- | ---------- |
+|bc1.z100.somedomain.com  10.100.10.1  | bc1.z200.somedomain.com  10.200.10.1 |
+bc2.z100.somedomain.com  10.100.10.2   | bc2.z200.somedomain.com  10.200.10.2 |
+bc3.z100.somedomain.com  10.100.10.3   | bc3.z200.somedomain.com  10.200.10.3 |
+fs1.z100.somedomain.com  10.100.20.1   | fs1.z200.somedomain.com  10.200.20.1 | 
+mq1.z100.somedomain.com  10.100.30.1   | mq1.z200.somedomain.com  10.200.30.1 |
+kz1.z100.somedomain.com  10.100.40.1   | kz1.z200.somedomain.com  10.200.40.1 |
+ka1.z100.somedomain.com  10.100.50.1   | ka1.z200.somedomain.com  10.200.50.1 |
+
+### Cluster Bigcouch
+This needs to be done before installing kazoo.
+
+On each Bigcouch node configure `z=2` in `/etc/kazoo/bigcouch/local.ini` as follows:
+```
+[cluster]  
+q=3  
+r=2  
+w=2  
+n=3  
+z=2  
+```
+
+Cluster together the bigcouch nodes from `bc1.z100` (in this example).
+
+```curl http://bc1.z100.somedomain.com:5986/nodes/bigcouch@bc1.z100.somedomain.com```  
+Returns  
+```{"_id":"bigcouch@bc1.z100.somedomain.com","_rev":"3-b13d076f367df4d0c52b236e654b836c"}```  
+
+Now add the zone to this bigcouch server and cluster together the other servers.
+```
+curl -X PUT bc1.z100.somedomain.com:5986/nodes/bigcouch@bc1.z100.somedomain.com -d '{"_rev":3-b13d076f367df4d0c52b236e654b836c", "zone":"z100"}'
+curl -X PUT bc1.z100.somedomain.com:5986/nodes/bigcouch@bc2.z100.somedomain.com -d '{"zone":"z100"}'
+curl -X PUT bc1.z100.somedomain.com:5986/nodes/bigcouch@bc3.z100.somedomain.com -d '{"zone":"z100"}'
```

Comment:
FYI this method can cause issues. The reason is that each bigcouch node creates a default node document for itself when it is started the first time. When you do this zone assignment at the same time as the creation of the node, sometimes this transaction silently conflicts, removing the zone value from the config.  

 The proper (and only safe) way to do this appears to be to create all the docs via put with empty -d '{}' payload, then update each doc (with the rev ID returned by this creation) to add the zone assignment. This is hard to catch up front and a bit painful to fix after the fact.